src/db/entity/ExecutionEntity.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  ManyToMany,
+  OneToMany,
+} from 'typeorm';
+import ConnectionEntity from './ConnectionEntity';
+import RuleEntity from './RuleEntity';
+
+/**
+ * A list of previous executions. Contains the test data, group, and test conditions.
+ * Currently supports only one database // TODO add support
+ */
+@Entity({ name: 'Execution' })
+class ExecutionEntity {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  timestamp: Date;
+
+  /**
+   * A list of the rules being tested
+   */
+  @OneToMany((_type) => RuleEntity, (rule) => rule)
+  rules: RuleEntity[];
```

Comment:
I think you'll need an on delete cascade here to delete all the rule groups once an execution is deleted?